keylime/ima/ima.py

```diff
@@ -42,7 +42,7 @@ class RUNTIME_POLICY_GENERATOR(enum.IntEnum):
 # the kernel maintainers decide to use a different algorithm for template-hash.
 EMPTY_RUNTIME_POLICY: RuntimePolicyType = {
     "meta": {
-        "version": RUNTIME_POLICY_CURRENT_VERSION,
+        "version": 1,  # Does not contain fields added after v1
```

Comment:
Should still be `RUNTIME_POLICY_CURRENT_VERSION`, and `rejects` should be added, shouldn't?  

Why we want to generate both v1 and v2 policy depending on the fields? This will make the schema evolution hard to track

Reply:
> Why we want to generate both v1 and v2 policy depending on the fields? This will make the schema evolution hard to track

The primary reason for this was that someone creating a v2 policy without a rejects entry or an empty reject map should still be able to use the policy with a previous version of keylime.

Reply:
I understand, yes. But how it is done now will make the future changes of the schema hard to maintain.

The json-schema has this "required" field to list the required elements, and "rejects" is not there in your change, so is optional and the parser will be OK with moving the version to 2 unconditionally

There is a cmd/convert_runtime_policy.py that will do the conversion for us, in direction 1->2 if required.

Reply:
I changed the version now in the EMPTY_RUNTIME_POLICY and added an empty rejects map. Wherever this EMPTY_RUNTIME_POLICY is now being used as a base policy and where the policy is written out later on (file or DB) I am now also calling ima.trim_runtime_policy() to get the version to the minimum that's required. If that's also undesirable I can drop it but you cannot use the policy on older versions of keylime anymore, even if there's no reason it is marked as a version 2 (because it uses no features of v2).

Reply:
Ah, I read this comment after writing mines, so I lacked the context provided here.

I think your goal should be kept, but I am not sure that the current approach is easy to maintain the the future. For me it is OK to move to v2 policy, even if is just a new optional field. Moving all to v2 and converting v1 to v2 should be OK too.

Let me check some bits in the codebase and I will try to comeback with a better comment.